mathesar_ui/src/components/filter-entry/FilterEntry.svelte

```diff
@@ -11,13 +11,15 @@
   import DynamicInput from '@mathesar/components/cell-fabric/DynamicInput.svelte';
   import { getDbTypeBasedInputCap } from '@mathesar/components/cell-fabric/utils';
   import { iconDelete } from '@mathesar/icons';
+  import { getTabularDataStoreFromContext } from '@mathesar/stores/table-data';
   import type { FilterEntryColumnLike } from './types';
   import { validateFilterEntry } from './utils';
 
   type T = $$Generic;
   type ColumnLikeType = FilterEntryColumnLike & T;
 
   const dispatch = createEventDispatcher();
+  const tabularData = getTabularDataStoreFromContext();
```

Comment:
This component is used in Data Explorer as well. Making it dependent on tabularData will break Data Explorer

Reply:
I do not know what's the best way to ensure such issues do not occur again. At the moment, please ensure you verify all the places a component is used when making changes to them.

Reply:
Thanks for catching this, @pavish. I've pushed changes to fix. 

> I do not know what's the best way to ensure such issues do not occur again

I think that you're also looking to improve this from a more systemic angle. 

I think it's worth considering reducing our dependency on Svelte's context feature or maybe seeking to eventually abandon it entirely. I see the usage of context as a major culprit in this case. Components should be encapsulated well enough that when I make changes within a component I _don't_ need to check everywhere that the component is used. Context is one of a few examples of where the encapsulation breaks down. This is certainly my fault for not being cognizant enough when I added that call to get context. I made an assumption. But I've been wondering if it might be better all-around for us to just do more prop drilling and less context passing. If we were to try reducing our usage of context, we could establish a rule that we only use context for values that we expect to be defined absolutely _everywhere_, like `toast`

Reply:
> I think it's worth considering reducing our dependency on Svelte's context feature or maybe seeking to eventually abandon it entirely. I see the usage of context as a major culprit in this case.

I don't think we should abandon context entirely. I think a better arrangement would be to never use external context in a common component i.e components in both the component-library and mathesar/components. I think it's okay to use contexts within page/route specific components.